smoke_tests/sns_list_subscriptions_test.go

```diff
@@ -0,0 +1,211 @@
+	"github.com/Admiral-Piett/goaws/app/models"
+	"github.com/Admiral-Piett/goaws/app/test"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/sns"
+
+	"github.com/stretchr/testify/assert"
+
+	af "github.com/Admiral-Piett/goaws/app/fixtures"
+
+	"github.com/gavv/httpexpect/v2"
+)
+
+func Test_List_Subscriptions_json_no_subscriptions(t *testing.T) {
+	server := generateServer()
+
+	defer func() {
+		server.Close()
+		test.ResetResources()
+	}()
+
+	sdkConfig, _ := config.LoadDefaultConfig(context.TODO())
+	sdkConfig.BaseEndpoint = aws.String(server.URL)
+	snsClient := sns.NewFromConfig(sdkConfig)
+
+	sdkResponse, err := snsClient.ListSubscriptions(context.TODO(), &sns.ListSubscriptionsInput{})
+
+	assert.Nil(t, err)
+	assert.Len(t, sdkResponse.Subscriptions, 0)
+}
+
+func Test_List_Subscriptions_json_multiple_subscriptions(t *testing.T) {
+	server := generateServer()
+	defaultEnv := app.CurrentEnvironment
+	conf.LoadYamlConfig("../app/conf/mock-data/mock-config.yaml", "NoQueueAttributeDefaults")
+	defer func() {
+		server.Close()
+		test.ResetResources()
+		app.CurrentEnvironment = defaultEnv
+	}()
+
+	sdkConfig, _ := config.LoadDefaultConfig(context.TODO())
+	sdkConfig.BaseEndpoint = aws.String(server.URL)
+	snsClient := sns.NewFromConfig(sdkConfig)
+
+	// add new topics to subscribe to
+	topicName := "new-topic-1"
+	createTopicResponse, _ := snsClient.CreateTopic(context.TODO(), &sns.CreateTopicInput{
+		Name: &topicName,
+	})
+	assert.Contains(t, *createTopicResponse.TopicArn, topicName)
+
+	topicName2 := "new-topic-2"
+	createTopicResponse2, _ := snsClient.CreateTopic(context.TODO(), &sns.CreateTopicInput{
+		Name: &topicName2,
+	})
+	assert.Contains(t, *createTopicResponse2.TopicArn, topicName2)
+
+	// subscribe to new topics
+	subscribeResponse, _ := snsClient.Subscribe(context.TODO(), &sns.SubscribeInput{
+		Protocol:              aws.String("sqs"),
+		TopicArn:              aws.String(fmt.Sprintf("%s:%s", af.BASE_SNS_ARN, topicName)),
+		Attributes:            map[string]string{},
+		Endpoint:              aws.String(fmt.Sprintf("%s:%s", af.BASE_SQS_ARN, "unit-queue1")),
+		ReturnSubscriptionArn: true,
+	})
+
+	subscribeResponse2, _ := snsClient.Subscribe(context.TODO(), &sns.SubscribeInput{
+		Protocol:              aws.String("sqs"),
+		TopicArn:              aws.String(fmt.Sprintf("%s:%s", af.BASE_SNS_ARN, topicName2)),
+		Attributes:            map[string]string{},
+		Endpoint:              aws.String(fmt.Sprintf("%s:%s", af.BASE_SQS_ARN, "unit-queue1")),
+		ReturnSubscriptionArn: true,
+	})
+
+	assert.NotNil(t, subscribeResponse)
+	assert.NotNil(t, subscribeResponse2)
+
+	app.SyncTopics.Lock()
+	defer app.SyncTopics.Unlock()
+
+	// check listed subscriptions
+	sdkResponse, err := snsClient.ListSubscriptions(context.TODO(), &sns.ListSubscriptionsInput{})
+	for _, sub := range sdkResponse.Subscriptions {
+		fmt.Println(*sub.TopicArn)

```

Comment:
Do we need this print, or even this loop for that matter?

Reply:
It wasn't required, unfortunately. Removed!